### PR TITLE
feat(ios): use applesimutils instead of fbsimctl for setLocation

### LINKED ISSUE #2736: (WIP) Replace fbsimctl with AppleSimUtils' setLocation
## Description
Integrate new location API in AppleSimUtils, and with that - eliminate the last usage of `fbsimctl`, and thus remove it as a dependency.

- [x] In the process, need to clean location related e2e test from specific refs to `fbsimctl`, and thus make it always runnable.
- [ ] Warn about running in headless mode (where setLocation doesn't work)

Related to #1371

### PR DESCRIPTION
**Pull request by noomorph**
## Description

- This pull request fixes #2736

In this pull request, I have replaced the implementation of `setLocation` for iOS to:

```
applesimutils --byId DEVICE_ID --setLocation "LAT, LON"
```

Besides, here I add two extra error handlers for improving developer experience:

1. `applesimutils` is not installed:

Screenshot 2021-07-20 at 15 57 12

```
detox[42391] ERROR: Detox failed to find "applesimutils" installed on the computer.
It is impossible to run tests on iOS simulators without this utility.

HINT: To install "applesimutils", run:
  brew tap wix/brew
  brew install applesimutils
```

2. `applesimutils` is outdated (for `--setLocation`):

image

```
  DetoxRuntimeError: Failed to set the location (-80.125, 66.5) on the device 6371830E-8B44-422B-A52C-DB0B62E6B6E3.
    Your current "applesimutils" version needs to be upgraded to 0.9.3 or higher.

    HINT: Try running:
      brew update && brew upgrade applesimutils
```

> _For features/enhancements:_
 - [x] I have added/updated the relevant references in the [documentation](https://github.com/wix/Detox/tree/master/docs) files.

### CODE REVIEW

**Comment by noomorph** [comment]
@d4vidi , Igor will come back tomorrow and update `applesimutils` on our CI. Then we'll re-check and we'll be able to merge.

File: docs/APIRef.DeviceObjectAPI.md [hunk 1]
```diff
@@ -320,7 +320,7 @@ Check out Detox's [own test suite.](../detox/test/e2e/06.device-orientation.test
 
 Sets the simulator/emulator location to the given latitude and longitude.
 
-> On iOS `setLocation` is dependent on [fbsimctl](https://github.com/facebook/idb/tree/4b7929480c3c0f158f33f78a5b802c1d0e7030d2/fbsimctl) which [is now deprecated](https://github.com/wix/Detox/issues/1371). If `fbsimctl` is not installed, the command will fail, asking for it to be installed.
+> On iOS `setLocation` is dependent on [applesimutils@546785](https://github.com/wix/AppleSimulatorUtils/commit/5467857407da4653305af119b2486051664a4f97) or more recent, that has `--setLocation` support.
```
Comment (d4vidi) [inline docs/APIRef.DeviceObjectAPI.md:323]: Should we be removing fbsimctl installation for any setup guide as well, then?

**Comment by d4vidi** [comment]
The test seems to fail on iOS. Any idea?
Anyways the equivalent test in Android is flaky, but please rerun to have it go green before merging 🙏🏻 .

File: docs/APIRef.DeviceObjectAPI.md (hunk 1 above)
Comment (noomorph) [reply docs/APIRef.DeviceObjectAPI.md:323]: There are no other mentions of `fbsimctl`, AFAIC.

**Comment by noomorph** [comment]
The issue is currently blocked by the fact that `applesimutils` has no release version since the `--setLocation` feature addition. Next week either I or Alon, someone, should take it and publish the new version.

Then, we can merge this feature.

Also, I am musing to add a check against outdated versions of `applesimutils`, so that users can see a meaningful hint message.

**Comment by d4vidi** [comment]
@noomorph So are there any blockers for releasing `applesimutils` 0.9.3?

**Comment by noomorph** [comment]
@d4vidi, currently none.

Waiting for @igorgn to bump `applesimutils` on our CI.

**Comment by d4vidi** [comment]
@alon-ha will get it done soon

**Comment by d4vidi** [comment]
UPDATE: Added setup guide refs to contextual error messages:
```
...
For complete details, refer to https://github.com/wix/Detox/blob/master/docs/Introduction.iOSDevEnv.md#install-applesimutils
```

**Comment by d4vidi** [comment]
Seems that even after upgrading `apple-sim-utils` to `0.9.4`, the setLocation test still fails on CI (iOS).

**Comment by d4vidi** [comment]
Here's a snippet of the device log from when this fails:
```
2021-08-01 04:43:33.473 Df example[48557:4f8e9] [com.apple.locationd.Core:Client] {"msg":"#CLLocationManager invoking #delegate - location unavailable", "self":"0x600002741110", "delegate":"0x600000b15e80", "selector":locationManager:didFailWithError:}
2021-08-01 04:43:33.473 Db example[48557:4f8e9] [com.apple.CoreAnalytics.event:send] com.apple.locationd.framework.CLLocationManager.delegate {"client":"com.wix.detox-example","delegateBundle":"com.wix.detox-example","delegateBundleIsMainBundle":true,"delegateClassName":"RNCGeolocation","delegateImage":"example","selector":"locationManager:didFailWithError:"}
```

The ultimate result is an this - which indicates the location is truly not available, on the device itself:

testFnFailure


I think the only reasonable way to continue with this is to work directly on a CI agent (once again - this _does_ work on my computer).

**Comment by d4vidi** [comment]
@noomorph this still requires attention, btw. Perhaps in bugathon week?

**Comment by noomorph** [comment]
Aloooon, help... 😭 @alon-ha 

TL;DR – it works locally but not on Jenkins. 🤷‍♂️

**Comment by alon-ha** [comment]
@d4vidi  @noomorph Guys I missed this PR with all the long holidays.. What needs to be done here which currently stuck us? Let's catch up on this tmw in the sync

**Comment by d4vidi** [comment]
@alon-ha looks like we need to grab a CI agent, log in and try to run it there and investigate the failures.

**Comment by asafkorem** [comment]
Locally, I had the same issue as we have on our CI, tried to fetch the location after granted the permissions and the error message was `Unable to retrieve location.`. 

However, I checked this with a simulator and applesimutils on my machine, and it failed to work also. Surprisingly it only worked after a few consecutive calls (>=2) for set-location, did the same on the tests and they also passed. Tried the same on the CI and it failed.

I asked @igorgn for a CI agent where I can debug this issue directly, it is also interesting why I experience the local issue (when only the second call for set-location works).

**Comment by asafkorem** [comment]
I looked on how applesimutils handles these calls but I haven’t found anything suspicious.

I tried to do the same with Xcode, and it seems to be exactly the same issue with Xcode (retrieving the location works only on the second setting). I suspect, that there might be some issue with the location module (or with how we work with it, I will look into it today).

However, this is only the local issue, the CI doesn't provide the location even after multiple calls to set it

https://user-images.githubusercontent.com/55082339/144189373-b256c12f-66e7-4b77-9cf0-2f6e1ae7f260.mov


https://user-images.githubusercontent.com/55082339/144189618-9908b9d7-10ad-4369-88e4-54360ea850ac.mov

.

**Comment by asafkorem** [comment]
This issue seems to be related: https://github.com/react-native-geolocation/react-native-geolocation/issues/139
There's also this potential fix: https://github.com/react-native-geolocation/react-native-geolocation/pull/168

**Comment by jonathanmos** [comment]
retest this please

**Comment by jonathanmos** [comment]
retest this please
